WWW/smoorthi-dev/scatter_gather/minidata_mutation_calling_v3.wdl

```diff
@@ -238,58 +246,68 @@ task MarkDuplicates {
     }
 
 
-
 task splitBambyChr {
-  input{
+  input {
     File bamtosplit
+    File baitosplit
     Array[String] chromosomes
-  }
-  
+    }
+    String base_file_name = basename(bamtosplit, "_final.duplicates_marked.bam")
   command <<<
-  set -eo pipefail
-  for x in "${chromosomes[@]}"
-  do
-  # Create BAM file
-  samtools view -b -@ 3 ~{bamtosplit} ${x} > ${x}.bam
-  # Create index file
-  samtools index ${x}.bam
-  done
-  >>>
-  output{
-    Array[File] bams = glob("*.bam")
-    Array[File] indexFiles = glob("*.bam.bai")
-  }
+    set -eo pipefail
+    baseFileName="~{base_file_name}"
   
+    for x in ~{sep=' ' chromosomes}; do
+    outputFile="${baseFileName}_${x}.bam"
+    samtools view -b -@ 3 ~{bamtosplit} $x > $outputFile
+
+    # Create index file
+    samtools index $outputFile
+    done
+    # List all bam and bai files created
+    ls *.bam > bam_list.txt
+    ls *.bam.bai > bai_list.txt
+  >>>
   
+  output {
+    Array[File] bams = read_lines("bam_list.txt")
+    Array[File] indexFiles = read_lines("bai_list.txt")
+  }
+
   runtime {
-    docker: "fredhutch/samtools:1.12"
+    docker: "fredhutch/bwa:0.7.17"
     cpu: 4
-    }
-    }
+  }
+  }
+
 
 task gatherBams {
     input {
         Array[File] bams
-        String sampleName
+        Array[String] sampleName
     }
+    # Use the first sample name to define the base name for merged files
+    String base_file_name = sampleName[0]
+
     command <<<
         set -eo pipefail
-        samtools merge -c -@3 ~{sampleName}.merged.bam ~{sep=" " bams}
-        samtools index ~{sampleName}.merged.bam
-        for bam in ~{sep=" " bams}
-        do
-            samtools index ~{bam}
-        done
+        baseFileName="~{base_file_name}"
+        samtools merge -c -@3 ${baseFileName}.merged.bam ~{sep=' ' bams}
+        samtools index ${baseFileName}.merged.bam
```

Comment:
You don't need to define baseFileName. Just use the WDL version and put quotes around the entire output file name.
```
samtools merge -c -@3 "~{base_file_name}.merged.bam" ~{sep=' ' bams}
samtools index "~{base_file_name}.merged.bam"
```

Reply:
So I tried that... For some reason in version 1.0 maybe? it doesn't resolveit inside the shell script!!!! Let me try that again